flask_monitor/metrics.py

```diff
@@ -79,27 +91,74 @@ def after_request(response):
         # pylint: enable=protected-access
         error_status = is_error(response.status_code)
         METRICS_REQUEST_LATENCY \
-            .labels("http", response.status_code, error_status, request.method, request.path) \
+            .labels("http", response.status_code, error_status, "", request.method, request.path) \
             .observe(request_latency)
         METRICS_REQUEST_SIZE.labels(
-            "http", response.status_code, error_status, request.method, request.path
+            "http", response.status_code, error_status, "", request.method, request.path
         ).inc(size_request)
         return response
+
     if error_fn is not None:
         is_error.__code__ = error_fn.__code__
     app.before_request(before_request)
     app.after_request(after_request)
+    return app, registry
 
-def watch_dependencies(dependency, func, time_execution=1500):
+def watch_dependencies(dependency, func, time_execution=1500, registry=None, app=current_app):
     """
     Register dependencies metrics
     """
+
+    if not registry:
+        registry = app.extensions.get("registry", CollectorRegistry())
+    app.extensions["registry"] = registry
+
+    # pylint: disable=invalid-name
+    DEPENDENCY_UP = Gauge(
+        'dependency_up',
+        'records if a dependency is up or down. 1 for up, 0 for down',
+        ["name"],
+        registry=registry
+    )
+
+    # pylint: disable=invalid-name
+    DEPENDENCY_UP_LATENCY = Histogram(
+        "dependency_request_seconds",
+        "records in a histogram the number of requests to dependency",
+        ["name", "type", "status", "isError", "errorMessage", "method", "addr"],
+        registry=registry
+    )
     def thread_function():
-        thread = threading.Timer(time_execution, lambda x: x + 1, args=(1,))
-        thread.start()
-        thread.join()
-        response = func()
-        DEPENDENCY_UP.labels(dependency).set(response)
-        thread_function()
-    thread = threading.Timer(time_execution, thread_function)
-    thread.start()
+        start = time.time()
+        try:
+            response = func()
+            DEPENDENCY_UP.labels(dependency).set(response)
+            DEPENDENCY_UP_LATENCY \
+                .labels(
+                    dependency, "http", "200" if response else "400",
```

Comment:
@Ziul, if `response` is a valid object, shouldn't you get the status code from the `response` object itself?

Reply:
This response is the return of the function. On the examples were normally a bool value, so there isn’t a status code. The documentation says about waiting for 0/1 value as this is an internal healthcheck.

Reply:
@Ziul, now I see the problem. Our referent lib implementation is [express-monitor](https://github.com/labbsr0x/express-monitor). From there we can see that the `DEPENDENCY_UP` is something we can help fill. But unfortunately, the same does not apply for the `DEPENDENCY_UP_LATENCY` metric. 

There instead, we make available an API to the developer, `collectDependencyTime`, and the placement logic and values are delegated to the developer using the lib. We should adopt the same in this lib.

Reply:
@abilioesteves I splitted the `watch_dependencies` in `watch_dependencies` and `collect_dependency_time`. Now it’s more similar to the implementation in [express-monitor](https://github.com/labbsr0x/express-monitor). Updated the tests and the example to reflect the changes too.

Now `watch_dependencies` only reflect the `DEPENDENCY_UP` metric. The `DEPENDENCY_UP_LATENCY` is updated with the `collect_dependency_time`.